src/commands/paranet/paranet-sync-command.js

```diff
@@ -0,0 +1,250 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable no-await-in-loop */
+import Command from '../command.js';
+import {
+    ERROR_TYPE,
+    PARANET_SYNC_FREQUENCY_MILLS,
+    OPERATION_ID_STATUS,
+    CONTENT_ASSET_HASH_FUNCTION_ID,
+    SIMPLE_ASSET_SYNC_PARAMETERS,
+    TRIPLE_STORE_REPOSITORIES,
+} from '../../constants/constants.js';
+
+class StartParanetSyncCommands extends Command {
+    constructor(ctx) {
+        super(ctx);
+        this.commandExecutor = ctx.commandExecutor;
+        this.blockchainModuleManager = ctx.blockchainModuleManager;
+        this.tripleStoreService = ctx.tripleStoreService;
+        this.ualService = ctx.ualService;
+
+        this.errorType = ERROR_TYPE.PARANET.PARANET_SYNC_ERROR;
+    }
+
+    async execute(command) {
+        const { commandOperationId, paranetId, tokenId } = command.data;
+
+        this.logger.info(
+            `Paranet sync: Starting paranet sync command for ${paranetId} with operation id: ${commandOperationId}, token id: ${tokenId}`,
+        );
+
+        const { blockchain, contract } = this.ualService.resolveUal(paranetId);
+        const assertionIds = await this.blockchainModuleManager.getLatestAssertionId(
+            blockchain,
+            contract,
+            tokenId,
+        );
+
+        // Go through all except the last one
+        for (let stateIndex = assertionIds.length - 2; stateIndex > 0; stateIndex -= 1) {
```

Comment:
Should loop condition be >0, if we have assertionIds.length equal to 2, would we want to get asset stateIndex 0?

Reply:
Or state index starts from 1?

Reply:
Initial code was [this](https://github.com/OriginTrail/ot-node/commit/f91a685b3ed5bda2a93ef0339967566544053088#diff-d798a8a22be1342ab3df626b595a30074f4c7ea46eaa834c3aab569a8f39344aR38), so I'm not sure

Reply:
Initial state index of the asset is 0, if there is one update it's assertionIds.length will be 2. When we call this command it won't get into the lope, as stateIndex will be 0 at beginning of the loop, and it should process this old asset and put it historical, but with current condition we won't do it?

Reply:
We should test this :D